pkgs/tools/typesetting/tex/texlive/default.nix

```
@@ -82,21 +81,19 @@ let
     }; # overrides
 
     in lib.mapAttrs mkTLPkg overridden;
-    # TODO: texlive.infra for web2c config?
-
 
   # create a TeX package: an attribute set { pkgs = [ ... ]; ... } where pkgs is a list of derivations
   mkTLPkg = pname: attrs:
     let
       version = attrs.version or (toString attrs.revision);
       mkPkgV = tlType: let
         pkg = attrs // {
-          sha512 = attrs.sha512.${tlType};
+          sha512 = attrs.sha512.${tlType} or attrs.sha512.run;
```

What is this needed for?

---

The `tlType == "tlpkg"` is not a true container, but a subset of the run container. The more self documenting code, also more robust probably, is:
```suggestion
          sha512 = attrs.sha512.${if tlType == "tlpkg" then "run" else tlType};
```

---

Updated with the `if tlType == "tlpkg"` code, evaluation has not changed.